torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1403,14 +1448,19 @@ def duck_int(self, val):
     # For convenience in testing, a source is allowed to be a str,
     # in which case we will assume it is a LocalSource
     #
+    # strict_mark_dyn lets you enforce stricter dynamic dim verification. When this flag is set,
+    # we assert that there are *no* constraints on the dim. If the flag is not set, we allow
+    # any constraining that allows for more than 2 values, aka, as long as it is not constrained to a single
+    # value.
+    #
     # simplified lets you omit duck sizing, equality and 0/1 guards.
     # This is useful for testing when you don't care about the boilerplate
     # guards, and it may be helpful for user output too (be careful though;
     # some equality guards are nontrivial!  It would be nice to get simplified
     # output to print them too).  It's private because it's not
     # intended for normal use
-    def produce_guards(self, placeholders, sources,
-                       source_ref=lambda n: n.name(), *, _simplified=False) -> List[str]:
+    def produce_guards(self, placeholders, sources, dynamic_ranges: Optional[Dict[int, "MinMaxConstraint"]],
```

Comment:
If you're going to accept None `dynamic_ranges`, might as well make this optional.

Reply:
I think the type signature on dynamic_ranges is wrong. It feels like you are Optional[List[Dict[int, MinMaxConstraint]]

Reply:
Yes